File: problems/problems_1432/Solution.java
package problems.problems_1432;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int maxDiff(int num) {
        String s = String.valueOf(num);
        int n = s.length();
        int max, min;
        int idx = 0;
        while (idx < n && s.charAt(idx) == '9') {
            idx++;
        }
        if (idx == n) {
            max = num;
        } else {
            max = Integer.parseInt(s.replace(s.charAt(idx), '9'));
        }
        if (s.charAt(0) == '1') {
            idx = 1;
            while (idx < n && (s.charAt(idx) == '0' || s.charAt(idx) == '1')) {
                idx++;
            }
            if (idx == n) {
                min = num;
            } else {
                min = Integer.parseInt(s.replace(s.charAt(idx), '0'));
            }
        } else {
            min = Integer.parseInt(s.replace(s.charAt(0), '1'));
        }
        return max - min;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int num = Integer.parseInt(inputJsonValues[0]);
        return JSON.toJSON(maxDiff(num));
    }
}

File: problems/problems_1432/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.maxDiff(test_input)

    def maxDiff(self, num: int) -> int:
        s = str(num)
        idx = 0
        while idx < len(s) and s[idx] == '9':
            idx += 1
        if idx == len(s):
            mx = num
        else:
            mx = int(s.replace(s[idx], '9'))
        if s[0] == '1':
            idx = 1
            while idx < len(s) and (s[idx] == '0' or s[idx] == s[0]):
                idx += 1
            if idx == len(s):
                mn = num
            else:
                mn = int(s.replace(s[idx], '0'))
        else:
            mn = int(s.replace(s[0], '1'))
        return mx - mn
